rust_src/src/buffers.rs

```diff
@@ -91,6 +96,19 @@ impl LispBufferRef {
     pub fn is_live(self) -> bool {
         LispObject::from_raw(self.name).is_not_nil()
     }
+
+    #[inline]
+    pub fn fetch_byte(&self, n: ptrdiff_t) -> u8 {
+        let base_addr =
+            if n  >= self.gpt_byte() {
+                self.gap_size()
+            } else {
+                0
+            };
+        let byte_addr = (base_addr + n +
+                         self.beg_addr() as ptrdiff_t - self.beg_byte()) as *const u8;
```

Comment:
looks like that should be `self.beg_addr().offset(base_addr + n - self.beg_byte())`